qcore/utils.py

```diff
@@ -6,8 +6,149 @@
 from shutil import rmtree
 import os
 import imp
+import yaml
+from collections import OrderedDict
+from collections import Mapping
 
-def setup_dir(directory, empty = False):
+
+class DotDictify(dict):
```

Comment:
you cannot use inbuilt functions as variables (dict)
https://docs.python.org/3.3/library/functions.html

Reply:
Not sure what it means. Could you please give an example

Reply:
ok, here it doesn't do what I think it does. you are extending the dictionary class.
my question now is why is this code so complicated when a quick search comes up with this:
https://www.blog.pythonlibrary.org/2014/02/14/python-101-how-to-change-a-dict-into-a-class/

Reply:
This class Dict2Obj does not work for nested dict like a={'b':2,'c':{'d':3,'z':4}}.
if you try aa=Dict2Obj(a); aa.c.d will throw an AttributeError

Reply:
why do we have a complete mess of code that uses nested dicts?? can we rather make nice code that reduces complexity rather than adding sticky tape everywhere and making everything complex. This whole file is full of trivial yet messy wrapper functions that really shouldn't be necessary and I honestly feel sick trying to understand this spaghetti. 
Ask someone else to review, I refuse to deal with this.